addon/mixins/in-viewport.js

```diff
@@ -121,16 +121,21 @@ export default Mixin.create({
       const boundingClientRect = element.getBoundingClientRect();
 
       if (boundingClientRect) {
-        this._triggerDidAccessViewport(
-          isInViewport(
-            boundingClientRect,
-            height,
-            width,
-            get(this, 'viewportTolerance')
-          )
+        let inView = isInViewport(
+          boundingClientRect,
+          height,
+          width,
+          get(this, 'viewportTolerance')
         );
-        if (get(this, 'viewportUseRAF')) {
-          rAFIDS[get(this, 'elementId')] = window.requestAnimationFrame(
+        this._triggerDidAccessViewport(inView);
+
+        if (!inView && get(this, 'viewportUseRAF')) {
+          let elementId = get(this, 'elementId');
+
+          if (rAFIDS[elementId]) {
+            window.cancelAnimationFrame(rAFIDS[elementId]);
+          }
+          rAFIDS[elementId] = window.requestAnimationFrame(
```

Comment:
This was constantly overwriting the previous request id and only the latest RAF ID was being canceled. Now the previous request will be canceled before making a new one.

Reply:
We have anywhere from 1 - 25 rAFs on a page.  Yes, I am seeing an rAF for each div.  I think a single rAF handler for all elements would be ideal. Right now all 25 rAFs are firing when the page loads and continues to fire until the divs come into view.  This fix was to address rAFs from continuing after everything is loaded.

Reply:
@jheth Wow, I've learned a bit about `cancelAnimationFrame`!

https://www.w3.org/TR/animation-timing/#dom-windowanimationtiming-cancelanimationframe

```The cancelAnimationFrame method is used to cancel a previously made request to schedule an animation frame update. When cancelAnimationFrame(handle) is called, the user agent must set the cancelled flag to true for the callback registered on this Document whose handle is handle. The cancelled flag is set whether the callback is in a animation frame request callback list or not. If there is no callback with the given handle, then this function does nothing.```

So apparently all this does is set a flag of `canceled` but doesn't remove the entry from the animation frame request callback list.  I've seen `cancelAnimationFrame` frequently used outside of the animating function, but I also guess it could happen from within.  Here is where it gets a bit muddy for me - In this case, we provided rAF a callback, are in the process of executing that callback and then we call `cancelAnimationFrame`, which...does it set the `canceled` flag or does it do nothing?  My gut based on the above is it is doing nothing.  ¯\_(ツ)_/¯ . What do you think?

Reply:
`cancelAnimationFrame`  has confused me. I tend to agree that it is doing nothing, but oddly we saw some benefit by adding this call.  I wonder if there's something else going on that prevented this recursive call from stopping after coming into view.  I was really thrown off when I saw this function keep ticking away.